### PR TITLE
matrix-commander: unstable-2021-08-05 -> 2.37.3

### PR DESCRIPTION
**Pull request by pacien**
###### Description of changes

The project switched to pyproject, allowing the use of
`buildPythonApplication` instead of `mkDerivation`.

Some of the modules listed as dependencies but shipping with Python
itself had to be patched out.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jian-lin** [comment]
Some tags, including 2.30.0, are deleted by the upstream.

Here are available tags now:

- v2.32.1
- v2.33.0
- v2.33.1
- v2.34.0
- v2.34.1
- v2.35.0

**Comment by pacien** [comment]
PR updated to 2.36.0, which was released in the meantime.

File: pkgs/applications/networking/instant-messengers/matrix-commander/default.nix [hunk 1]
```diff
@@ -1,38 +1,63 @@
...
+      -e '/uuid/d' \
+      -e '/argparse/d' \
+      -e '/asyncio/d' \
+      -e '/datetime/d' \
+      setup.cfg requirements.txt
 
-    runHook postInstall
+    # Dependencies not correctly detected
+    sed -i \
+      -e '/dbus-python/d' \
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/matrix-commander/default.nix:42]: Isn't this just dbus?

Comment (pacien) [reply pkgs/applications/networking/instant-messengers/matrix-commander/default.nix:42]: It's listed as `dbus-python` in
https://github.com/8go/matrix-commander/blob/7cd410247d64a9b202e2f01c4375339b07ef80ab/requirements.txt#L6
and
https://github.com/8go/matrix-commander/blob/7cd410247d64a9b202e2f01c4375339b07ef80ab/setup.cfg#L37

I tried adding both `dbus` and `dbus-python` as build inputs, but this results in the following error (when not using `sed` to remove it from `requirements.txt`):

```
Executing pipInstallPhase
/build/source/dist /build/source
Processing ./matrix_commander-2.37.3-py3-none-any.whl
Requirement already satisfied: python-magic in /nix/store/l19ka7yqf5cvb38vnb2h7ivh36lgwz85-python3.9-python-magic-0.4.26/lib/python3.9/site-packages (from matrix-commander==2.37.3) (0.4.26)
Requirement already satisfied: markdown in /nix/store/fkr0fvyd9vh0fddp0cxmlf0ziqji9pb1-python3.9-markdown-3.3.7/lib/python3.9/site-packages (from matrix-commander==2.37.3) (3.3.7)
ERROR: Could not find a version that satisfies the requirement dbus-python (from matrix-commander) (from versions: none)
ERROR: No matching distribution found for dbus-python
```

**Pull request merged**